js/map.js

```diff
@@ -75,6 +94,7 @@ window.map = (function () {
     if (dragged) {
       mapPinsMainElement.addEventListener('click', clickPreventDefaultHandler);
     }
+    window.debounce(applyFilter);
```

Comment:
You don't need to debounce here. It's a single event.

Reply:
yes, you're right, but when is a small delay with receiving data from the server then my applyFilter function will not get data on time

Reply:
Debounce should not solve this problem. It's a hack and accident that it's working

Reply:
Why do you need to filter data on mouse up?

Reply:
You data should be rendered in the callback you pass to the request.

Reply:
done